crates/formatter/src/format_token_macro.rs

```diff
@@ -0,0 +1,73 @@
+/// The macro `format_tokens` is a convenience macro to
+/// use when writing a list of tokens that should be at the same level
+/// without particular rule.
+///
+/// # Examples
+///
+/// Let's suppose you need to create tokens for the string `"foo": "bar"`,
+/// you would write:
+///
+/// ```rust
+/// use rome_formatter::{format_tokens, FormatToken};
+/// let token = format_tokens!("foo", ":", FormatToken::Space, "bar");
+/// ```
+///
+/// The macro can be also nested, although the macro needs to be decorated with the token you need.
+/// For example, let's try to format following string:
+///
+/// ```no_rust
+/// "foo": {
+///   "bar": "lorem"
+/// }
+/// ```
+/// You would write it like the following:
+///
+/// ```rust
+/// use rome_formatter::{format_tokens, IndentToken, LineToken, FormatToken};
+/// let token = format_tokens!(
+///   "foo",
+///   ":",
+///   IndentToken::new(
+///     format_tokens!(LineToken::hard(), "bar", ":", FormatToken::Space, "lorem")
+///   ),
+///   "}"
+/// );
+///
+/// let inner = FormatToken::concat(vec![
+///     FormatToken::from(LineToken::hard()),
+///     FormatToken::string("bar"),
+///     FormatToken::string(":"),
+///     FormatToken::Space,
+///     FormatToken::string("lorem")
+/// ]);
+/// let outer = FormatToken::concat(vec![
+///     FormatToken::string("foo"),
+///     FormatToken::string(":"),
+///     FormatToken::indent(inner),
+///     FormatToken::string("}")
+/// ]);
+///
+/// assert_eq!(token, outer);
```

Comment:
It might be clearer to reader what's happening here if you assert against the formatted string.
```suggestion
/// 
/// assert_eq!(r#"foo: {
  bar: lorem
}"#, format_token(&token).code());
```

It also guarantees that the created tokens match what you expect (missing opening brackets after `foo`, missing quotes).